test/Hydrate/HydrateUsingReflectionTest.php

```diff
@@ -49,4 +50,18 @@ public function it_ignores_extra_keys_in_the_data_array()
 
         $this->assertSame('bar', $object->foo());
     }
+
+    /**
+     * @test
+     */
+    public function it_works_for_objects_without_properties()
+    {
+
+        $object = new ClassWithoutProperties();
+
+        $hydrate = new HydrateUsingReflection();
+
+        // This class doesn't have any property which should be no problem
+        $hydrate->hydrate(['extra' => 'no problem'], $object);
```

Comment:
To be extra clear about what's going on, I think an empty array would be better here.